src/main/java/org/prgrms/prgrmsspring/repository/voucher/FileVoucherRepository.java

```diff
@@ -0,0 +1,111 @@
+
+@Slf4j
+@Profile("prod")
+@Repository
+public class FileVoucherRepository implements VoucherRepository {
+    private final Map<UUID, Voucher> store = new ConcurrentHashMap<>();
+    private final String filePath;
+
+
+    public FileVoucherRepository(@Value("${file.path.voucher}") String filePath) {
+        this.filePath = filePath;
+        readAndStoreFile();
+    }
+
+    @PreDestroy
+    private void saveFile() {
+        writeFile();
+    }
+
+    private void readAndStoreFile() {
+        try {
+            FileInputStream inputStream = new FileInputStream(filePath);
+            List<String> fileStrings = readFile(inputStream);
+            List<Voucher> vouchers = convertFileStringsToObjectLists(fileStrings);
+            vouchers.forEach(v -> store.put(v.getVoucherId(), v));
+        } catch (IOException e) {
+            throw new NotFoundException(ExceptionMessage.NOT_FOUND_FILE.getMessage());
+        }
+    }
+
+    private List<String> readFile(InputStream inputStream) throws IOException {
+        List<String> lines = new ArrayList<>();
+        try (BufferedReader reader = new BufferedReader(new InputStreamReader(inputStream))) {
+            String line;
+            while ((line = reader.readLine()) != null) {
+                lines.add(line);
+            }
+        }
+        return lines;
+    }
+
+
+    public List<Voucher> convertFileStringsToObjectLists(List<String> fileStrings) {
+        List<Voucher> voucherList = new ArrayList<>();
+        fileStrings.stream().map(s -> s.split(",")).forEach(split -> {
+            String type = split[0];
+            UUID uuid = UUID.fromString(split[1]);
+            long value = Long.parseLong(split[2]);
+            VoucherType voucherType = VoucherType.of(type);
+            Voucher voucher = voucherType.constructVoucher(uuid, value);
+            voucherList.add(voucher);
+        });
+        return voucherList;
+    }
+
+
+    public void writeFile() {
+        try (FileOutputStream outputStream = new FileOutputStream(filePath)) {
+            List<String> fileStrings = convertObjectListsToFileStrings(store.values());
+            for (String str : fileStrings) {
+                outputStream.write(str.getBytes());
+                outputStream.write(System.lineSeparator().getBytes());
+            }
+            outputStream.flush();
+        } catch (IOException e) {
+            throw new NotFoundException(ExceptionMessage.NOT_FOUND_FILE.getMessage());
+        }
+    }
+
+    public List<String> convertObjectListsToFileStrings(Collection<Voucher> voucherList) {
+        List<String> fileStrings = new ArrayList<>();
+        voucherList.forEach(voucher -> {
+            String fileString = "," + voucher.getVoucherId() + "," + voucher.getAmount();
+            VoucherType voucherType = VoucherType.of(voucher);
+            fileString = voucherType.getName() + fileString;
+            fileStrings.add(fileString);
+        });
+        return fileStrings;
+    }
```

Comment:
```suggestion
    public List<String> convertObjectListsToFileStrings(Collection<Voucher> voucherList) {
        return voucherList.stream()
                .map(voucher -> VoucherType.of(voucher).getTitle() + voucher.getVoucherId() + "," + voucher.getAmount())
                .toList();
    }
```

한번에 처리가 가능해보이는데 맞을까요?

Reply:
네 맞습니다. 해당 내용으로 수정하겠습니다